### PR TITLE
III-3585 Add Sentry decorator to PSR loggers

### PR DESCRIPTION
**Pull request by bertramakers**
### Added

- Added Sentry decorator to PSR loggers, similar to https://github.com/cultuurnet/udb3-search-service/pull/79 and https://github.com/cultuurnet/udb3-silex/pull/519

---

Ticket: https://jira.uitdatabank.be/browse/III-3585

### CODE REVIEW

File: src/SentryPsrLoggerDecorator.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+     */
+    private $logger;
+
+    public function __construct(SentryErrorHandler $sentryHandler, LoggerInterface $logger)
+    {
+        $this->sentryHandler = $sentryHandler;
+        $this->logger = $logger;
+    }
+
+    public function error($message, array $context = array()): void
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:34]:
```suggestion
    public function error($message, array $context = []): void
```

File: src/SentryPsrLoggerDecorator.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $this->sendThrowableToSentry($context);
+        $this->logger->error($message, $context);
+    }
+
+    public function emergency($message, array $context = array()): void
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:40]:
```suggestion
    public function emergency($message, array $context = []): void
```

File: src/SentryPsrLoggerDecorator.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $this->sendThrowableToSentry($context);
+        $this->logger->emergency($message, $context);
+    }
+
+    public function alert($message, array $context = array()): void
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:46]:
```suggestion
    public function alert($message, array $context = []): void
```

File: src/SentryPsrLoggerDecorator.php [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $this->logger->alert($message, $context);
+    }
+
+    public function critical($message, array $context = array()): void
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:51]:
```suggestion
    public function critical($message, array $context = []): void
```

File: src/SentryPsrLoggerDecorator.php [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $this->logger->critical($message, $context);
+    }
+
+    public function warning($message, array $context = array()): void
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:56]:
```suggestion
    public function warning($message, array $context = []): void
```

File: src/SentryPsrLoggerDecorator.php [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $this->logger->warning($message, $context);
+    }
+
+    public function notice($message, array $context = array()): void
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:61]:
```suggestion
    public function notice($message, array $context = []): void
```

File: src/SentryPsrLoggerDecorator.php [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $this->logger->notice($message, $context);
+    }
+
+    public function info($message, array $context = array()): void
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:66]:
```suggestion
    public function info($message, array $context = []): void
```

File: src/SentryPsrLoggerDecorator.php [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $this->logger->info($message, $context);
+    }
+
+    public function debug($message, array $context = array()): void
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:71]:
```suggestion
    public function debug($message, array $context = []): void
```

File: src/SentryPsrLoggerDecorator.php [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $this->logger->debug($message, $context);
+    }
+
+    public function log($level, $message, array $context = array()): void
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:76]:
```suggestion
    public function log($level, $message, array $context = []): void
```

File: src/SentryPsrLoggerDecorator.php [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $this->logger->log($level, $message, $context);
+    }
+
+    private function sendThrowableToSentry(array $context = array()): void
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:81]:
```suggestion
    private function sendThrowableToSentry(array $context = []): void
```

File: src/SentryPsrLoggerDecorator.php [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $throwable = $context['exception'] ?? null;
+        if (!$throwable || !($throwable instanceof Throwable)) {
+            return;
+        }
+
+        $this->sentryHandler->handle($throwable);
```
Comment (jdrieghe) [inline src/SentryPsrLoggerDecorator.php:88]: Let's not overcomplicate an implementation to force an early return in a void method

```suggestion
        if ($throwable instanceof Throwable) {
            $this->sentryHandler->handle($throwable);
        }
```

**Review by jdrieghe — CHANGES_REQUESTED** [review]

File: src/SentryPsrLoggerDecorator.php (hunk 11 above)
Comment (bertramakers) [reply src/SentryPsrLoggerDecorator.php]: Yeah you're right, it's just an automatic habit at this point. Which imho is useful in 99% of cases but indeed not here.

**Review by jdrieghe — APPROVED** [review]

**Pull request merged**